Docs/Architecture.md

```diff
@@ -0,0 +1,1429 @@
+# OneCrypto Architecture
+
+> **⚠️ Platform Support Notice:**  
+> This architecture is currently **X64 ONLY**. AArch64 platforms require a different model due to architectural differences in StandaloneMm and firmware volume handling. See [Platform and Architecture Considerations](#platform-and-architecture-considerations) for details.
+
+## Overview
+
+OneCrypto is a refactoring of the MuCryptoRelease, MuSharedCrypto, and CryptoBinary concepts. This effort is focused on simplifying and improving industry adoption, reliability, and compatibility. The goal is to help all parties shipping products with cryptographic support do it better and stay up to date more easily.
+
+### Repositories
+
+- **`MuCryptoRelease`** - Repository and project for building the crypto binary and carrying crypto library dependencies like OpenSSL
+- **`UEFI-MuBaseCore-Edk2`** - Generic representation of the "fork" of common code that is common in UEFI industry, providing the interface for crypto and other common services
+
+### Key Benefits
+
+1. **Simplified Updates**: Make it much easier to update crypto for many platforms and stay current with minimal validation
+2. **Reduced Build Cost**: No need to download and compile OpenSSL for every platform build
```

Comment:
I'd reduce directly referencing OpenSSL throughout the document since the design is intended to allow crypto providers to be swapped out.

```suggestion
2. **Reduced Build Cost**: No need to download and compile a crypto provider like OpenSSL for every platform build
```